### PR TITLE
Added Cycle Detection in a Directed Graph in Java

### PR DESCRIPTION
**Pull request by YASH01009**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

(WoC)

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#1468 Directed Graph Cycle Detection / WoC)

### CODE REVIEW

File: Java/README.md [hunk 1]
```diff
@@ -140,7 +141,7 @@ _add list here_
 ## Sorting
 
 - [BeadSort](sort/BeadSort.java)
-- [Bubble Sort](sort/BubbleSort.java)
+- [Bubble Sort](sort/BubbleSort.ja
```
Comment (HarshCasper) [inline Java/README.md]: I don't understand this change? Can you revert this to the original hyperlink?

File: Java/graphs/DirectedCycleDetection.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/**
+Cycle detection in a directed graph
+using dfs. Here, ststus array shows if
```
Comment (HarshCasper) [inline Java/graphs/DirectedCycleDetection.java]: Typo Errors like `ststus`. You can try fixing this.

File: Java/graphs/DirectedCycleDetection.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		dfs(graph, v, status);
+		if(cycle == false)
+		    System.out.println("No cycle exits in the given graph");
+	    else
+		    System.out.println("Cycle exists in the given graph");
+	}
+
+	public static void main(String[] args) {
+		cycle = false;
+		int v, e, a, b;
```
Comment (HarshCasper) [inline Java/graphs/DirectedCycleDetection.java]: Can you try using more meaningful variable names?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Code looks great to me! 

@plazzy99 and @iamrajiv Would love you guys taking a look at it

**Comment by YASH01009** [comment]
All the requested changes are now done !!

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@YASH01009 Change the variable name `v` to `vertex` and `e` to `edge`.

**Comment by YASH01009** [comment]
@iamrajiv changes done !!

**Review by iamrajiv — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Comment by iamrajiv** [comment]
@all-contributors please add @YASH01009 for code

**Pull request merged**